Assets/MRTK/SDK/Features/Input/Handlers/ObjectManipulator.cs

```diff
@@ -867,39 +924,61 @@ private void ApplyTargetTransform(MixedRealityTransform targetTransform)
                 }
                 if (elasticTypes.HasFlag(TransformFlags.Scale))
                 {
                     HostTransform.localScale = scaleElastic.ComputeIteration(targetTransform.Scale, Time.deltaTime);
                 }
                 else
                 {
-                    HostTransform.localScale = smoothingActive ? Smoothing.SmoothTo(HostTransform.localScale, targetTransform.Scale, scaleLerpTime, Time.deltaTime) : targetTransform.Scale;
+                    HostTransform.localScale = isSmoothing ? Smoothing.SmoothTo(HostTransform.localScale, targetTransform.Scale, scaleLerpTime, Time.deltaTime) : targetTransform.Scale;
                 }
             }
             else
             {
-                rigidBody.velocity = ((1f - Mathf.Pow(moveLerpTime, Time.deltaTime)) / Time.deltaTime) * (targetTransform.Position - HostTransform.position);
+                // There is a RigidBody. Potential different paths for near vs far manipulation
+                if (isNearManipulation && !useForcesForNearManipulation)
+                {
+                    // This is a near manipulation and we're not using forces
+                    // Apply direct updates but account for smoothing
 
-                var relativeRotation = targetTransform.Rotation * Quaternion.Inverse(HostTransform.rotation);
-                relativeRotation.ToAngleAxis(out float angle, out Vector3 axis);
-                if (angle > 180f)
-                    angle -= 360f;
-                if (axis.IsValidVector())
+                    if (isSmoothing)
+                    {
+                        rigidBody.MovePosition(Smoothing.SmoothTo(rigidBody.position, targetTransform.Position, moveLerpTime, Time.deltaTime));
+                        rigidBody.MoveRotation(Smoothing.SmoothTo(rigidBody.rotation, targetTransform.Rotation, rotateLerpTime, Time.deltaTime));
+                    }
+                    else
+                    {
+                        rigidBody.MovePosition(targetTransform.Position);
+                        rigidBody.MoveRotation(targetTransform.Rotation);
+                    }
+                }
+                else
                 {
-                    rigidBody.angularVelocity = ((1f - Mathf.Pow(rotateLerpTime, Time.deltaTime)) / Time.deltaTime) * (axis.normalized * angle * Mathf.Deg2Rad);
+                    // We are using forces
+
+                    rigidBody.velocity = ((1f - Mathf.Pow(moveLerpTime, Time.deltaTime)) / Time.deltaTime) * (targetTransform.Position - HostTransform.position);
+
+                    var relativeRotation = targetTransform.Rotation * Quaternion.Inverse(HostTransform.rotation);
+                    relativeRotation.ToAngleAxis(out float angle, out Vector3 axis);
+                    if (angle > 180f)
+                        angle -= 360f;
```

Comment:
Super nit: Bring this into if (axis.IsValidVector()) because it's not needed unless IsValidVector ends up being true.

Reply:
Though these lines show changed, they're only "changed" because I rearranged the call order somewhat to clarify operations. These are original ObjectManipulator lines of code. So while I agree with the nit (and do realize it's just a nit), hopefully we can maybe address it after the PR is closed. :)

Reply:
I just fixed this, closing this out.